### PR TITLE
Added IsNullOrEmptyConverter_Tests

### LINKED ISSUE #176: [Enhancement] HasData Converter
Hi, this is a converter I use in several projects, you have 2 similar converters in this project already -
- IsNullOrEmptyConverter 
- ListIsNullOrEmptyConverter 

You can pass any object into the and it would return true/ false whether it has data or not.
Let me know if you'd like a PR for this, I understand if not as you already have 2 similar converters already -

```csharp
public class HasDataConverter : IValueConverter
{
    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
    {
        switch (value)
        {
            case null:
                return false;
            case string x:
                return !string.IsNullOrWhiteSpace(x);
            case IList list:
                return list.Count > 0;
            default:
                return true;
        }
    }

    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
    {
        throw new NotImplementedException();
    }
}
```

### PR DESCRIPTION
**Pull request by IeuanWalker**
### Description of Change ###
Bring more unit tests

### API Changes ###
None

### Behavioral Changes ###
None

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

**Comment by IeuanWalker** [comment]
@sthewissen @jfversluis 
I havnt added a check for lists as there are already 2 converted that cover it 'ListIsNullOrEmptyConverter' and 'ListIsNotNullOrEmptyConverter'.

But makes sense to me to add the list check into this converter to and remove the 2 converters that are specific to lists. Only behaviour change with that would be the list converter currently throws an ArgumentException if the parameter isn't and IEnumerable or null.

Let me know what you think :)

File: XamarinCommunityToolkit/Converters/IsNotNullOrEmptyConverter.shared.cs [hunk 1]
```diff
@@ -18,8 +18,13 @@ public class IsNotNullOrEmptyConverter : ValueConverterExtension, IValueConverte
...
 		/// <returns>A boolean indicating if the incoming value is not null and not empty.</returns>
-		public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
-			=> value is string str ? !string.IsNullOrWhiteSpace(str) : value != null;
+		public object Convert(object value, Type targetType, object parameter, CultureInfo culture) =>
+			value switch
+			{
+				null => false,
+				string x => !string.IsNullOrWhiteSpace(x),
+				_ => true,
+			};
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Converters/IsNotNullOrEmptyConverter.shared.cs:27]: What is the difference between the old code and the new one?)

**Review by AndreiMisiukevich — CHANGES_REQUESTED** [review]
```\|/```

File: XamarinCommunityToolkit/Converters/IsNullOrEmptyConverter.shared.cs [hunk 2]
```diff
@@ -18,8 +18,13 @@ public class IsNullOrEmptyConverter : ValueConverterExtension, IValueConverter
...
 		/// <returns>A boolean indicating if the incoming value is null or empty.</returns>
-		public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
-			=> value == null || (value is string str && string.IsNullOrWhiteSpace(str));
+		public object Convert(object value, Type targetType, object parameter, CultureInfo culture) =>
+			value switch
+			{
+				null => true,
+				string x => string.IsNullOrWhiteSpace(x),
+				_ => false,
+			};
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Converters/IsNullOrEmptyConverter.shared.cs:27]: This code performs exactly the same check

**Comment by IeuanWalker** [comment]
@AndreiMisiukevich sorry ye your right. Based on the conversation in the issue and a quick look at the code I thought it if it wasn't a string it was defaulting to false.

I can update this PR so it just has the added tests `IsNullOrEmptyConverter_Tests`, unless you want me to merge the list converters into these converters.

**Comment by AndreiMisiukevich** [comment]
> @AndreiMisiukevich sorry ye your right. Based on the conversation in the issue and a quick look at the code I thought it if it wasn't a string it was defaulting to false.
> 
> I can update this PR so it just has the added tests `IsNullOrEmptyConverter_Tests`, unless you want me to merge the list converters into these converters.

I think more tests are always welcome 👍

File: XamarinCommunityToolkit.UnitTests/Converters/IsNullOrEmptyConverter_Tests.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace Xamarin.CommunityToolkit.UnitTests.Converters
+{
+	public class IsNullOrEmptyConverter_Tests
+	{
+		[Theory]
+		[InlineData("Test", false)]
+		[InlineData(null, true)]
+		[InlineData(typeof(IsNullOrEmptyConverter), false)]
+		[InlineData("", true)]
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit.UnitTests/Converters/IsNullOrEmptyConverter_Tests.cs:13]:
```suggestion
		[InlineData(null, true)]
		[InlineData("", true)]
		[InlineData("Test", false)]
		[InlineData(typeof(IsNullOrEmptyConverter), false)]
```

File: XamarinCommunityToolkit.UnitTests/Converters/IsNotNullOrEmptyConverter_Tests.cs [hunk 4]
```diff
@@ -9,8 +9,9 @@ public class IsNotNullOrEmptyConverter_Tests
 		[Theory]
 		[InlineData("Test", true)]
 		[InlineData(null, false)]
+		[InlineData(typeof(IsNotNullOrEmptyConverter), true)]
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit.UnitTests/Converters/IsNotNullOrEmptyConverter_Tests.cs:12]:
```suggestion
		[InlineData(typeof(IsNotNullOrEmptyConverter), true)]
		[InlineData(null, false)]
```

**Review by AndreiMisiukevich — APPROVED** [review]

**Review by bartlannoeye — APPROVED** [review]

**Pull request merged**